### DISCUSSION TITLE
True stress/true strain or engineering stress/engineering strain

### DISCUSSION THREAD

**Discussion by xchengood** [idaholab/moose#19434] [Q&A Modules: Solid mechanics] [answered]
Dear Moose experts or users,

1. Does the following inputting syntax output true stress/true strain or engineering stress/engineering strain?
![image](https://user-images.githubusercontent.com/74226455/142665209-a0456990-8cc8-4aeb-b54a-0b8b95d9bb05.png)
One example is here (https://github.com/idaholab/moose/blob/next/modules/tensor_mechanics/test/tests/crystal_plasticity/stress_update_material_based/update_method_test.i).

I found the demonstration for the 'strain = FINITE' and 'add_variables = true' (https://mooseframework.inl.gov/modules/tensor_mechanics/Strains.html). But I don't figure out what is exact for the 'generate_output = stress_zz'. Will it output true stress?

2. I suppose this example (https://mooseframework.inl.gov/source/materials/crystal_plasticity/CrystalPlasticityHCPDislocationSlipBeyerleinUpdate.html) just be implemented on a single element. How can I run it for the polycrystal?

Thank you.

**Comment by jiangwen84** [comment, marked as answer]
In general, MOOSE uses true strain and true stress (Cauchy stress). The `generate_output=stress_zz` will output cauchy stress in the zz direction ((2,2) component in the 3x3 stress tensor)

For CP, you can assign different Euler angles to different blocks/elements for polycrystalline modeling.
